crates/bevy_math/src/bezier.rs

```diff
@@ -0,0 +1,263 @@
+    /// across the length of the curve from t = `0..=1`, and sample with the supplied
+    /// `sample_function`.
+    #[inline]
+    pub fn sample(&self, subdivisions: i32, sample_function: fn(&Self, f32) -> P) -> Vec<P> {
+        (0..=subdivisions)
+            .map(|i| {
+                let t = i as f32 / subdivisions as f32;
+                sample_function(self, t)
+            })
+            .collect()
+    }
+
+    /// Split the Bezier curve into `subdivisions` evenly spaced `t` values across the length of the
+    /// curve from t = `0..=1`. sampling the position at each step.
+    pub fn to_positions(&self, subdivisions: i32) -> Vec<P> {
+        self.sample(subdivisions, Self::position)
+    }
+
+    /// Split the Bezier curve into `subdivisions` evenly spaced `t` values across the length of the
+    /// curve from t = `0..=1`. sampling the velocity at each step.
+    pub fn to_velocities(&self, subdivisions: i32) -> Vec<P> {
+        self.sample(subdivisions, Self::velocity)
+    }
+
+    /// Split the Bezier curve into `subdivisions` evenly spaced `t` values across the length of the
+    /// curve from t = `0..=1` . sampling the acceleration at each step.
+    pub fn to_accelerations(&self, subdivisions: i32) -> Vec<P> {
+        self.sample(subdivisions, Self::acceleration)
+    }
+}
+
+/// A 2-dimensional Bezier curve used for easing in animation.
+///
+/// A cubic Bezier easing curve has control point `p0` at (0, 0) and `p3` at (1, 1), leaving only
+/// `p1` and `p2` as the remaining degrees of freedom. The first and last control points are fixed
+/// to ensure the animation begins at 0, and ends at 1.
+#[derive(Default, Clone, Copy, Debug, PartialEq)]
+pub struct CubicBezierEasing {
+    /// Control point P1 of the 2D cubic Bezier curve. Controls the start of the animation.
+    pub p1: Vec2,
+    /// Control point P2 of the 2D cubic Bezier curve. Controls the end of the animation.
+    pub p2: Vec2,
+}
+
+impl CubicBezierEasing {
+    /// Construct a cubic bezier curve for animation easing, with control points `p1` and `p2`.
+    /// These correspond to the two free "handles" of the bezier curve.
+    ///
+    /// This is a very common tool for animations that accelerate and decelerate smoothly. For
+    /// example, the ubiquitous "ease-in-out" is defined as `(0.25, 0.1), (0.25, 1.0)`.
+    pub fn new(p1: impl Into<Vec2>, p2: impl Into<Vec2>) -> Self {
+        Self {
+            p1: p1.into(),
+            p2: p2.into(),
+        }
+    }
+
+    /// Maximum allowable error for iterative bezier solve
+    const MAX_ERROR: f32 = 1e-7;
+
+    /// Maximum number of iterations during bezier solve
+    const MAX_ITERS: u8 = 8;
+
+    /// Given a `time` within `0..=1`, returns an eased value within `0..=1` that follows the cubic
+    /// Bezier curve instead of a straight line.
+    ///
+    /// The start and endpoints will match: ease(0) = 0 and ease(1) = 1.
+    pub fn ease(&self, time: f32) -> f32 {
```

Comment:
This uses `time`, while the others use `t`. Mild preference for `t`, but I only actually care about consistency.

Reply:
No. Time is the x-axis of the Bezier. The reason easing is complex is that you need to take this `x` value, solve for the corresponding `t`, then compute `y`.

Reply:
Added docs